third_party/amd/backend/compiler.py

```diff
@@ -169,24 +169,15 @@ def make_ttgir(mod, metadata, options):
         passes.ttgpuir.add_remove_layout_conversions(pm)
         amd.passes.ttgpuir.add_optimize_epilogue(pm)
         passes.ttgpuir.add_optimize_dot_operands(pm, True)
-        use_new_pipeliner = os.getenv("TRITON_HIP_USE_NEW_STREAM_PIPELINE", "1") == "1"
         if amd.has_matrix_core_feature(options.arch):
-            if use_new_pipeliner:
-                # In the old pipeliner we only support num_stages = 0/1, which means something
-                # different than the NVIDIA side. In the new pipeliner we unify the num_stages
-                # interpretation. Default to use 2 stages if not explicitly set.
-                num_stages = options.num_stages if options.num_stages != 0 else 2
-                amd.passes.ttgpuir.add_stream_pipelinev2(pm, num_stages)
-            else:
-                if options.num_stages == 0:
-                    amd.passes.ttgpuir.add_stream_pipeline(pm)
+            num_stages = options.num_stages if options.num_stages != 0 else 2
```

Comment:
We should also drop this too--it was meant to smooth transition. After v1 gone we need to respect what the user set--if it's 0 then we don't pipeline anymore.

Reply:
This will have a big influence on almost everyone. For now, all gemm-like kernels on AMD backend uses num_stages=0. Removing it will cause a significant perf regression.

Reply:
Yeah `num_stages=0` just need to be fixed in all downstream users--we cannot be beholden to that.